cli/command/container/tty.go

```diff
@@ -34,19 +34,48 @@ func resizeTtyTo(ctx context.Context, client client.ContainerAPIClient, id strin
 	}
 
 	if err != nil {
-		logrus.Debugf("Error resize: %s", err)
+		logrus.Debugf("Error resize: %s\r", err)
 	}
+	return err
 }
 
-// MonitorTtySize updates the container tty size when the terminal tty changes size
-func MonitorTtySize(ctx context.Context, cli command.Cli, id string, isExec bool) error {
-	resizeTty := func() {
-		height, width := cli.Out().GetTtySize()
-		resizeTtyTo(ctx, cli.Client(), id, height, width, isExec)
-	}
+// resizeTty is to resize the tty with cli out's tty size
+func resizeTty(ctx context.Context, cli command.Cli, id string, isExec bool) error {
+	height, width := cli.Out().GetTtySize()
+	return resizeTtyTo(ctx, cli.Client(), id, height, width, isExec)
+}
 
-	resizeTty()
+// safeResizeTty is to resize the tty's size safely
+func safeResizeTty(ctx context.Context, cli command.Cli, id string, isExec bool, resizeTtyFunc func(ctx context.Context, cli command.Cli, id string, isExec bool) error) {
+	rttyFunc := resizeTtyFunc
+	if rttyFunc == nil {
+		rttyFunc = resizeTty
+	}
+	err := rttyFunc(ctx, cli, id, isExec)
+	if err != nil {
+		retryTimes := 5
```

Comment:
The loop looked a bit complicated to me; wondering; would this work? (pls double-check; it's just a very quick draft)

```go
go func() {
	var err error
	for retry := 0; retry < 5; retry++ {
		time.Sleep(10 * time.Millisecond)
		err = rttyFunc(ctx, cli, id, isExec)
		if err == nil {
			break
		}
	}
	if err != nil {
		fmt.Fprintln(cli.Err(), "Resize tty error, we'll use a default size. You can resize the tty size manually.")
	}
}()
```

Reply:
In the beginning after I wrote this code, I had the same feeling as you. But I have used it for about half a year, it works fine for my team.
But I can't promise it is the perfect solution, if some one have more simple way, please tell us, thanks. @kolyshkin Please take a look and help me to check this code, thank you very much.

Reply:
I will test this code.

Reply:
Congratulations, this quick draft works very well.